libraries/AP_ICEngine/AP_ICEngine.h

```diff
@@ -73,7 +73,7 @@ class AP_ICEngine {
 
     static AP_ICEngine *get_singleton() { return _singleton; }
 
-private:
+//!!private:
```

Comment:
Yeah, no.  This isn't acceptable.

If you think you can make a reasonable case for exposing the state machine state, make an accessor for it.  Check other available interface lements first....